### ISSUE TITLE
Мысли, пожелания по модулю

### ISSUE THREAD

**Issue by pchelk1n** [kohana-pack/Pagination#1]
Есть модуль https://github.com/kloopko/kohana-pagination
Не хватает переопределения шаблона с использованием другого шаблонизатора.
Что еще ?

**Comment by aktuba** [comment]
Раскрой мысль, plz. Что значит "переопределения шаблона с использованием другого шаблонизатора"?
По мне, если используется шаблонизатор, он должен подменять собой класс View. Иначе, при смене шаблонизатора на готовом проекте придется весь код править.

**Comment by pchelk1n** [comment]
Согласен. Но для тех кто не переопределяет View, такой функционал был бы полезен. Правда пока еще не думал над реализацией.

**Comment by biakaveron** [comment]
Вероятно, таким людям следует просто переопределить метод Pagination::render() под свои нужды. Для бОльшего удобства можно вынести все, что завязано на рендеринг шаблона, в отдельный метод _render(). То есть в render() останутся проверки, общие для любого шаблонизатора, а в _render() можно будет переопределить чисто свои хотелки.

**Comment by pchelk1n** [comment]
Ага согласен. Чего тогда не хватает в том модуле ?

**Comment by pchelk1n** [comment]
Посмотрел пагинацию в Yii. В шаблонах используют вывод через виджет.

**Comment by biakaveron** [comment]
Это простой, несложный модуль. Не надо к нему прикручивать никаких лишних наворотов. ИМХО, на данный момент вполне достаточно довести до ума kohana-pagination (версии для 3.2 и 3.3), актуализировать стандартные шаблоны для пагинатора (взять за основу самые популярные + какие-нить привычные из соцсетей + подсмотреть в других фреймах).

Сперва нужна основа - просто работающий модуль. Далее будем собирать фидбеки, нужно будет что-то допилить - пользователи сами скажут об этом.

**Comment by Fer0x** [comment]
Приветствую. Касаемо Pagination обычно все что включает релизация этого модуля - вывод самой пагинации. Ни у кого не было идеи чтобы этот модуль так же выполнял работу с данными? Т.е. на входе получал array($orm, $per_page, $current_page) и возвращал array($results, $total_pages)

**Comment by Fer0x** [comment]
Еще один момент, на текущий момент чтобы kohana-pagination корректно работал, необходимо чтобы для был отдельный роут для текущего контроллера. В админке с общем роутом под неё, kohana-pagination берет только первый сегмент url - /admin/.

**Comment by biakaveron** [comment]
> Ни у кого не было идеи чтобы этот модуль так же выполнял работу с данными? 

Зачем завязываться на определенный модуль (который еще и не является обязательным)? Тогда уж пусть модель ORM подготавливает данные для пагинатора.

> Еще один момент, на текущий момент чтобы kohana-pagination корректно работал, необходимо чтобы для был отдельный роут для текущего контроллера. 

Можно поподробнее? А лучше - в отдельный тикет.

**Comment by Fer0x** [comment]
Да, на счет ORM согласен.

Еще предложение ввести число максимального кол-во дополнений, для того чтобы на последнюю страницу не приходилось 1-2 результата. Подробнее тут: http://habrahabr.ru/qa/3198/

**Comment by biakaveron** [comment]
Идею понял. Тикет, please ))

**Comment by Fer0x** [comment]
> Зачем завязываться на определенный модуль (который еще и не является обязательным)? Тогда уж пусть модель ORM подготавливает данные для пагинатора.
> И все таки по поводу этой штуки. Как это лучше организовать, как отдельный класс наследующий ORM в рамках модуля?

**Comment by biakaveron** [comment]
Можно сделать абстрактную модель ORM, от нее наследовать свои Model_Article, Model_News и тд
